lib/FusionInventory/Agent/Task/Inventory/Linux/Storages.pm

```diff
@@ -115,6 +129,31 @@ sub _getDevices {
     return @devices;
 }
 
+# get serial & firmware numbers from hdparm, if available
+sub _getHdparmInfo {
+    my (%params) = @_;
+
+    return unless _correctHdparmAvailable(%params);
```

Comment:
To follow previous comment and permit memoize usage:
```suggestion
    return unless _correctHdparmAvailable(
        root => $params{root},
        dump => $params{dump},
    );
```
You should even know these 2 arguments can only be set during testing, see t/tasks/inventory/linux/storages.t.

Reply:
Since 'device' key was different each call, memoize didn't work. Thank you.